db/migrate/20180622140220_add_recurring_logic_to_sync_plan.rb

```diff
@@ -0,0 +1,6 @@
+class AddRecurringLogicToSyncPlan < ActiveRecord::Migration[5.1]
+  def change
+    add_column :katello_sync_plans, :foreman_tasks_recurring_logic_id, :integer
+    add_foreign_key :katello_sync_plans, :foreman_tasks_recurring_logics, :name => "katello_sync_plan_foreman_tasks_recurring_logic_fk", :column => "foreman_tasks_recurring_logic_id"
+  end
```

Comment:
I would go ahead and loop over all the sync plans, create their Recurring logic, and then set the foreman_tasks_recurring_logic_id to non-nullable